rasa/shared/utils/validation.py

```diff
@@ -168,12 +168,17 @@ def validate_yaml_schema(yaml_file_content: Text, schema_path: Text) -> None:
 
     try:
         c.validate(raise_exception=True)
-    except SchemaError:
+    except SchemaError as e:
+        if e not in c.errors:
```

Comment:
Are these the same kind of errors? Are they even comparable?

Reply:
I couldn't find much about SchemaError in `pykwalify` documentation, but in their library they're the same kind (`c.errors` is a list of SchemaErrors)

Reply:
I double-checked and you're right @joejuzl , there is a nuance, `c.errors` is a list of `SchemaError.SchemaErrorEntry` which is an inner class. 

I can check `e.msg` against each message in `c.errors` to prevent duplication but it doesn't seem to cause any issues in our code if I add the `SchemaError` in the list of validation_errors alongside `c.errors` 🤔

I might have to change the type annotation to Union of SchemaError and SchemaErrorEntry in the `YamlValidationException`. WDYT?

Reply:
I think it was fine how it was to begin with tbh. As we know what they are not duplicate now.

Reply:
I've actually kept the check at `msg` level because in some cases there might be duplicates based on what I could understand from `pykwalify` codebase - for example when the `SchemaError` that gets raised during validation includes the `c.errors` messages rather than it being a custom SchemaError that Rasa introduces through pykwalify extensions. I'm going to have to edit the check (reverse it) based on this insight.